docs/source/guide/build_pipelines/pipeline_parametrization.rst

```diff
@@ -0,0 +1,135 @@
+Parametrization
+===============
+
+Oftentimes, data scientists/engineers need to run the same pipeline with different parameters.
+For instance, they may want to use a different data set for model training and/or prediction.
+To produce a parametrized pipeline, we can use pipeline API's (optional) ``input_parameters`` argument.
+
+As a concrete example, consider the :ref:`pipeline created in the Basics section <iris_pipeline_module>`,
+where we got an "inflexible" pipeline that has the data source (``url``) as a fixed value rather than a tunable parameter:
+
+.. code-block:: python
+    :emphasize-lines: 8, 19
+
+    # ./output/pipeline_basics/iris_pipeline_module.py
+
+    import pandas as pd
+    from sklearn.linear_model import LinearRegression
+
+
+    def get_iris_preprocessed():
+        url = "https://raw.githubusercontent.com/LineaLabs/lineapy/main/examples/tutorials/data/iris.csv"
+        df = pd.read_csv(url)
+        color_map = {"Setosa": "green", "Versicolor": "blue", "Virginica": "red"}
+        df["variety_color"] = df["variety"].map(color_map)
+        df["d_versicolor"] = df["variety"].apply(lambda x: 1 if x == "Versicolor" else 0)
+        df["d_virginica"] = df["variety"].apply(lambda x: 1 if x == "Virginica" else 0)
+        return df
+
+    [...]
+
+    def run_session_including_iris_preprocessed():
+        # Given multiple artifacts, we need to save each right after
+        # its calculation to protect from any irrelevant downstream
+        # mutations (e.g., inside other artifact calculations)
+        import copy
+
+        artifacts = dict()
+        df = get_iris_preprocessed()
+        artifacts["iris_preprocessed"] = copy.deepcopy(df)
+        mod = get_iris_model(df)
+        artifacts["iris_model"] = copy.deepcopy(mod)
+        return artifacts
+
+    [...]
+
+    if __name__ == "__main__":
+        # Edit this section to customize the behavior of artifacts
+        artifacts = run_all_sessions()
+        print(artifacts)
+
+Instead, we could have run
+
+.. code-block:: python
+   :emphasize-lines: 10
+
+    # Load artifacts to use in pipeline building
+    preprocessing_art = lineapy.get("iris_preprocessed")
+    modeling_art = lineapy.get("iris_model")
+
+    # Build an Airflow pipeline using artifacts
+    lineapy.to_pipeline(
+        pipeline_name="iris_pipeline_parametrized",
+        artifacts=[preprocessing_art.name, modeling_art.name],
+        dependencies={modeling_art.name: {preprocessing_art.name}},
+        input_parameters=["url"],  # Specify variable(s) to parametrize
+        output_dir="./output/pipeline_parametrization/",
+        framework="AIRFLOW",
+    )
+
+to get a parametrized pipline, like so:
+
+.. code-block:: python
+   :emphasize-lines: 9, 20, 42
+
+    # ./output/pipeline_parametrization/iris_pipeline_parametrized_module.py
+
+    import argparse
+
+    import pandas as pd
+    from sklearn.linear_model import LinearRegression
+
+
+    def get_iris_preprocessed(url):
```

Comment:
we dont have a default value like with run_session_including_iris_preprocessed?

Reply:
No, we don't for these modularized functions. I think the intention behind this design (feel free to add/correct @mingjerli) is to have these functions as a more general reusable components free from specificity such as default param values; but give such specificity for actual execution of these functions (e.g., `run_session_including_xxx()`, `run_all_sessions()`).